### PR TITLE
Prow update master

### PR DESCRIPTION
**Pull request by TerryHowe**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Update release-tools according to https://github.com/kubernetes-csi/csi-release-tools/issues/7

**Which issue(s) this PR fixes**:
Related https://github.com/kubernetes-csi/csi-release-tools/pull/259
Closes: https://github.com/kubernetes-csi/csi-proxy/pull/353

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: release-tools/prow.sh [hunk 1]
```diff
@@ -86,7 +86,7 @@ configvar CSI_PROW_BUILD_PLATFORMS "linux amd64 amd64; linux ppc64le ppc64le -pp
 # which is disabled with GOFLAGS=-mod=vendor).
 configvar GOFLAGS_VENDOR "$( [ -d vendor ] && echo '-mod=vendor' )" "Go flags for using the vendor directory"
 
-configvar CSI_PROW_GO_VERSION_BUILD "1.22.5" "Go version for building the component" # depends on component's source code
+configvar CSI_PROW_GO_VERSION_BUILD "1.23.1" "Go version for building the component" # depends on component's source code
```
Comment (mauriciopoppe) [inline release-tools/prow.sh:89]: There's this recent issue that's related with the bump to go 1.23 https://github.com/kubernetes-csi/csi-proxy/issues/361.

Hopefully the tests pass as usual but if not we might need to make additional changes to go.mod to make it work.

**Review by mauriciopoppe — COMMENTED** [review]
Thanks for the PR! I left a comment that we'll figure out after we see tests running.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**